src/Core/Error.js

```diff
@@ -1,115 +1,107 @@
+		return propagateErrorBoundary(host, host.host, err, exception, origin, signature)
 
-	if (typeof printErr === 'function')
-		return printErr(err)
+	if (exception.bubbles && printErrorException(exception))
+		throw exception.error
 }
 
 /**
- * @param {*} value
- * @return {Object}
+ * @param {Element} element
+ * @param {Element?} host
+ * @param {*} err
+ * @param {Exception} exception
+ * @param {string} origin
+ * @param {number} signature
+ * @param {Component} owner
+ * @param {boolean?}
  */
-function getErrorDescription (value) {
-	return {enumerable: true, configurable: true, value: value}
+function recoverErrorBoundary (element, host, err, exception, origin, signature, owner) {
+	switch (origin) {
+		case SharedComponentWillMount:
+		case SharedGetInitialState:
+			getLifecycleOnce(owner, SharedSiteRender, noop)
+		case SharedSiteRender:
+		case SharedComponentWillUnmount:
+			break
+		case SharedGetChildContext:
+		case SharedComponentShouldUpdate:
+		case SharedComponentWillUpdate:
+		case SharedComponentWillReceiveProps:
+			element.active && getLifecycleOnce(owner, SharedSiteRender, noop)
+		default:
+			element.active && replaceErrorChildren(element, getElementDefinition())
+	}
+
+	if (owner && !owner[SymbolException] && owner[SharedComponentDidCatch])
+		return owner[SymbolException] = getLifecycleBoundary(element, SharedComponentDidCatch, owner[SymbolException] = err, exception)
+
+	return isValidElement(host) && isValidElement(host.host)
 }
 
 /**
  * @param {Element} element
- * @param {Error} error
- * @param {string} from
+ * @param {Element} snapshot
  */
-function getErrorException (element, error, from) {
-	if (!(error instanceof Error))
-		return getErrorException(element, new Error(error), from)
-
-	var componentStack = ''
-	var tabs = '    '
-	var host = element
-
-	while (host && host.type) {
-		componentStack += tabs + '<' + getDisplayName(host.type) + '>\n'
-		tabs += '  '
-		host = host.host
-	}
-
-	var errorMessage = 'The above error occurred in `\n' + componentStack + '` from "' + from + '"'
-	var errorStack = error.stack + '\n\n' + errorMessage
+function replaceErrorChildren (element, snapshot) {
+	reconcileElement(element.id === SharedElementComponent ? element.children : element, snapshot)
+}
 
-	return defineProperties(error, {
-		errorLocation: getErrorDescription(from),
-		errorStack: getErrorDescription(errorStack),
-		errorMessage: getErrorDescription(errorMessage),
-		componentStack: getErrorDescription(componentStack),
-		defaultPrevented: getErrorDescription(false),
-		preventDefault: getErrorDescription(function () {
-			return !!defineProperty(error, 'defaultPrevented', getErrorDescription(true))
-		}),
-		inspect: getErrorDescription(function () {
-			return errorStack
-		})
-	})
+/**
+ * @param {(object|string)} exception
+ */
+function printErrorException (exception) {
+	try { console.error(exception.toString()) } catch (e) {} finally { return exception }
```

Comment:
`catch` is optional with `finally`

Reply:
I thought that was the case as well but found out that these two produce different results.

```js
try { throw '' } finally {}
```

throws.

```js
try { throw '' } catch (e) {} finally {}
```
shallows the error

Reply:
or `console`/`console.error` might not exist/might have been subbed to one that throws etc.

Reply:
There is [optional `catch` syntax proposal](https://github.com/tc39/proposal-optional-catch-binding) in stage 3, if you are interested in setting it up in your Babel config: 

```js
try { throw '' } catch {} finally {}
```